feathr_project/feathr/registry/_feathr_registry_client.py

```diff
@@ -0,0 +1,631 @@
+import importlib
+import json
+import logging
+import os
+from pathlib import Path
+import sys
+from urllib.parse import urlparse
+from uuid import UUID
+from azure.identity import DefaultAzureCredential
+from typing import Any, Dict, List, Optional, Tuple
+from re import sub
+from jinja2 import Template
+
+import requests
+from feathr.constants import INPUT_CONTEXT, TYPEDEF_ANCHOR, TYPEDEF_ANCHOR_FEATURE, TYPEDEF_DERIVED_FEATURE, TYPEDEF_SOURCE
+from feathr.definition.anchor import FeatureAnchor
+from feathr.definition.dtype import FeatureType, str_to_value_type, value_type_to_str
+from feathr.definition.feature import Feature
+from feathr.definition.feature_derivations import DerivedFeature
+from feathr.definition.repo_definitions import RepoDefinitions
+from feathr.definition.source import HdfsSource, InputContext, JdbcSource, Source
+from feathr.definition.transformation import ExpressionTransformation, Transformation, WindowAggTransformation
+from feathr.definition.typed_key import TypedKey
+from feathr.registry.feature_registry import FeathrRegistry
+from feathr.utils._file_utils import write_to_file
+
+
+def to_camel(s):
+    if not s:
+        return s
+    if isinstance(s, str):
+        if "_" in s:
+            s = sub(r"(_)+", " ", s).title().replace(" ", "")
+            return ''.join([s[0].lower(), s[1:]])
+        return s
+    elif isinstance(s, list):
+        return [to_camel(i) for i in s]
+    elif isinstance(s, dict):
+        return dict([(to_camel(k), s[k]) for k in s])
+    
+class _FeatureRegistry(FeathrRegistry):
+    def __init__(self, project_name: str, endpoint: str, project_tags: Dict[str, str] = None, credential=None, config_path=None):
+        self.project_name = project_name
+        self.project_tags = project_tags
+        self.endpoint = endpoint
+        self.credential = DefaultAzureCredential(
+            exclude_interactive_browser_credential=False) if credential is None else credential
+        self.project_id = None
+
+    def register_features(self, workspace_path: Optional[Path] = None, from_context: bool = True, anchor_list=[], derived_feature_list=[]):
+        """Register Features for the specified workspace. 
```

Comment:
Specify type hint in the method parameters for `anchor_list` and `derived_feature_list`